app/vmui/packages/vmui/src/hooks/useSearchParamsFromObject.ts

```diff
@@ -7,9 +7,16 @@ const useSearchParamsFromObject = () => {
   const [searchParams, setSearchParams] = useSearchParams();
 
   const setSearchParamsFromKeys = useCallback((objectParams: Record<string, string | number>) => {
-    const hasSearchParams = !!Array.from(searchParams.values()).length;
+    const hasSearchParams = !!searchParams.size;
     let hasChanged = false;
 
+    searchParams.keys().forEach(key => {
```

Comment:
It will be better to not mutate `searchParams` here, because it can lead to unpredictable behavior of render react components. So, I would suggest to create new object from the `searchParams` and mutate it.